mypy/server/astdiff.py

```diff
@@ -83,7 +83,10 @@ def is_similar_node_shallow(n: SymbolTableNode, m: SymbolTableNode) -> bool:
                 nn.is_newtype == mn.is_newtype and
                 is_same_mro(nn.mro, mn.mro))
     if isinstance(n.node, Var) and isinstance(m.node, Var):
-        return is_identical_type(n.node.type, m.node.type)
+        if n.node.type is None and m.node.type is None:
```

Comment:
This first `if` seems redundant? It shouldn't have any effect on the binder (since it can't guarantee that either of those is not None) and if both types are None the next `if` definitely won't trigger, so the final `return True` will be executed.